.github/workflows/build.yml

```diff
@@ -0,0 +1,90 @@
+name: build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  release:
+    types:
+      - released
+  workflow_dispatch:
+
+jobs:
+  cmake:
+    strategy:
+      matrix:
+        runs-on:
+          - ubuntu-latest
+          # macos-13 is an intel runner, macos-14 is apple silicon
+          - macos-13
+          - macos-latest
+      fail-fast: false
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: zacharyburnett/setup-abseil-cpp@de39f445295c887839e30c864ffbbb1c0231bc83 # 1.0.5
+        with:
+          cmake-build-args: "-DCMAKE_CXX_STANDARD=17 -DABSL_PROPAGATE_CXX_STD=ON -DABSL_ENABLE_INSTALL=ON -DBUILD_TESTING=off -DCMAKE_POSITION_INDEPENDENT_CODE=ON"
+          abseil-version: "20240722.0"
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - run: mkdir build
+      - run: cmake -DCMAKE_CXX_STANDARD=17 -DCMAKE_PREFIX_PATH=/usr/local/ -DBUILD_TESTS=OFF ..
+        working-directory: build/
+      - run: sudo cmake --build . --parallel --target=install
+        working-directory: build/
+  python-wheels:
+    strategy:
+      matrix:
+        runs-on:
+          - ubuntu-latest
+          # macos-13 is an intel runner, macos-14 is apple silicon
+          - macos-13
+          - macos-latest
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.13"
+        exclude:
+          # importing the package from the wheel instantly segfaults on macOS Python <= 3.12
+          - runs-on: macos-latest
+            python-version: "3.8"
+          - runs-on: macos-latest
+            python-version: "3.9"
+          - runs-on: macos-latest
+            python-version: "3.10"
+          - runs-on: macos-latest
+            python-version: "3.11"
+          - runs-on: macos-latest
+            python-version: "3.12"
+          - runs-on: macos-13
+            python-version: "3.8"
+          - runs-on: macos-13
+            python-version: "3.9"
+          - runs-on: macos-13
+            python-version: "3.10"
+          - runs-on: macos-13
+            python-version: "3.11"
+          - runs-on: macos-13
+            python-version: "3.12"
+      fail-fast: false
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: zacharyburnett/setup-abseil-cpp@de39f445295c887839e30c864ffbbb1c0231bc83 # 1.0.5
+        with:
+          cmake-build-args: "-DCMAKE_CXX_STANDARD=17 -DABSL_PROPAGATE_CXX_STD=ON -DABSL_ENABLE_INSTALL=ON -DBUILD_TESTING=off -DCMAKE_POSITION_INDEPENDENT_CODE=ON"
+          abseil-version: "20240722.0"
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: pip install build
```

Comment:
Specifically for the macos wheel there will need to be a `delocate` step. What was the issue with cibuildwheel? Their step includes delocation. Also a test step that installs the wheel itself. Which is where I encountered issues.

Reply:
This might be the case for other Linux based distros.

Reply:
Why is `delocate` needed? If I understand correctly, it's to remove external shared libs. Which ones would it remove?

Reply:
It's not removing them, it's actually bundling them in the wheel and updates any references to point to the new location.

I have had issues sharing a wheel built on macos arm using a python from pyenv because there were harcoded paths with my username.

Searching for a solution pointed me to delocate.

Reply:
as far as why I stopped using `pypa/cibuildwheel`, I found that it was difficult to include the installed `abseil-cpp` in `cibuildwheel`'s container, so I figured it would be easier to replicate the steps used in `cibuildwheel`, which includes `pypa/build`, instead: https://github.com/google/s2geometry/pull/390#issuecomment-2468585295

Reply:
As an example, when I run `wheel unpack` on the built wheel and then `vim` the `_s2geometry.so` binary. Among the random symbols I see things like 
![image](https://github.com/user-attachments/assets/e9bf5125-6d99-4c33-9c21-79a8007d60d8)

which are hardcoded paths that show specific dependencies on python and libssl install locations.

The result of `delocate-wheel` used by `cibuildwheel` is a wheel with `.dylibs` directory. The new `_s2geometry.so` has library references like 
![image](https://github.com/user-attachments/assets/94b1c1e6-9c8f-4757-a07f-90787bba4cbc)

Reply:
I'm surprised there would be hardcoded absolute paths in the output, but ok.

Reply:
It might be worth revisiting using the `cibuildwheel` action and attempting to insert `abseil-cpp` into the containers it spins up; as far as I can tell, `cibuildwheel` uses the `manylinux` container and also does macOS emulation but using `libc++` instead of `libSystem`